metashare/api/jobs.py

```diff
@@ -0,0 +1,96 @@
+from cumulusci.core.config import BaseProjectConfig
+from cumulusci.core.runtime import BaseCumulusCI
+from django.utils.text import slugify
+from django_rq import job
+from github3 import login
+from github3.exceptions import UnprocessableEntity
+
+from .sf_scratch_orgs import extract_owner_and_repo, make_scratch_org
+
+
+class MetadeployProjectConfig(BaseProjectConfig):
+    def __init__(
+        self, *args, repo_root=None, repo_url=None, commit_ish=None, **kwargs
+    ):  # pragma: nocover
+
+        owner, repo_name = extract_owner_and_repo(repo_url)
+
+        repo_info = {
+            "root": repo_root,
+            "url": repo_url,
+            "name": repo_name,
+            "owner": owner,
+            "commit": commit_ish,
+        }
+
+        super().__init__(*args, repo_info=repo_info, **kwargs)
+
+
+class MetaDeployCCI(BaseCumulusCI):
+    project_config_class = MetadeployProjectConfig
+
+
+def create_scratch_org(*, scratch_org, user, repo_url, commit_ish):
+    url = make_scratch_org(user, repo_url, commit_ish)
+    scratch_org.url = url
+    scratch_org.save()
+
+
+def try_to_make_branch(repository, *, new_branch, base_branch):
+    branch_name = new_branch
+    counter = 0
+    while True:
+        try:
+            latest_sha = repository.branch(base_branch).latest_sha()
+            repository.create_branch_ref(branch_name, latest_sha)
+            return branch_name
+        except UnprocessableEntity:
+            counter += 1
+            branch_name = f"{new_branch}-{counter}"
+
+
+def create_branches_on_github(*, user, repo_url, project, task):
+    gh = login(token=user.gh_token)
+    owner, repo = extract_owner_and_repo(repo_url)
+    repository = gh.repository(owner, repo)
+
+    # Make project branch:
+    project_branch_name = slugify(project.name)
```

Comment:
This should also get a prefix (defaults to "feature/"), pulled from `cumulusci.yml` on the repo.

Reply:
This turns out to occasion a valuable and worthwhile but large-ish refactor. I'm hoisting the GitHub checkout up above each of the other tasks, so it's available at this point where we need it, too, without repeatedly doing the checkout.